var/lib/delphix-platform/os-migration

```diff
@@ -28,6 +28,19 @@ function die() {
 }
 
 function perform_migration() {
+	#
+	# On Illumos rpool/crashdump has its mountpoint set to "legacy",
+	# and we mount it manually. On Linux we want it to be mounted
+	# automatically by setting its mountpoint property.
+	# During migration this would be done by the delphix-platform service
+	# on first boot however that fails since the kdump-tools service
+	# creates a file under /var/crash before the delphix-platform can be
+	# run. To mitigate this problem we set the mountpoint here, as we
+	# run before kdump-tools.
+	#
+	zfs set mountpoint=/var/crash rpool/crashdump ||
```

Comment:
With this scheme, when will `/var/crash` be mounted in the boot process in relation to when the root filesystem itself gets mounted? I'm wondering if we have risk of losing crash dumps if the system crashes before it gets mounted... Would an alternative be to not have a separate dataset for `/var/crash` at all, and make it be part of the root filesystem (only trying to understand the constraints)?

Reply:
IIRC, the reason we have a separate dataset is so it can have a unique quota. So, if we don't use a dataset, we'll need to determine if we need a quota, and if so how to do that without ZFS.

Reply:
The reason why we have a separate dataset in `/var/crash` is because we want to set a quota there. I don't know well enough how the crash kernel works to figure out what would happen if we panic before this dataset is mounted, I think this would need more investigations.

Reply:
I don't know enough about this to raise an issue with this change, this was just a thought that came to mind. It makes me wonder if, for example, we panic in early boot in ZFS, how would we diagnose this?